channel.go

```diff
@@ -0,0 +1,340 @@
+package cli
+
+import (
+	"encoding/json"
+	"time"
+
+	stream "github.com/GetStream/stream-chat-go/v5"
+	"github.com/cheynewallace/tabby"
+	"github.com/urfave/cli/v2"
+)
+
+func NewChannelCmd(config *Config) *cli.Command {
+	return &cli.Command{
+		Name:        "channel",
+		Usage:       "Manage channels.",
+		Description: "Manage channels.",
+		Subcommands: []*cli.Command{
+			getChannelCmd(config),
+			newListChannelsCmd(config),
+			newCreateChannelCmd(config),
+			newDeleteChannelCmd(config),
+			newUpdateChannelCmd(config),
+		},
+	}
+}
+
+func getChannelCmd(config *Config) *cli.Command {
+	return &cli.Command{
+		Name:        "get",
+		Usage:       "Get a channel by channel type and channel name.",
+		Description: "Get a channel by channel type and channel name.",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:     "channel-type",
+				Aliases:  []string{"t"},
+				Usage:    "The type of the channel. Such as 'messaging' or 'livestream'.",
+				Required: true,
+			},
+			&cli.StringFlag{
+				Name:     "channel-name",
+				Aliases:  []string{"n"},
+				Usage:    "The name of the channel.",
+				Required: true,
+			},
+			&cli.BoolFlag{
+				Name:     "json",
+				Usage:    "Print the raw JSON representation of the API response.",
+				Required: false,
+			},
+		},
+		Action: func(ctx *cli.Context) error {
+			r, err := getOrCreateChannel(config, ctx)
+
+			if err != nil {
+				return err
+			}
+
+			t := tabby.New()
+			t.AddHeader("CID", "Member count", "Created By", "Created At", "Updated At", "Extra Data")
+			t.AddLine(r.Channel.CID,
+				r.Channel.MemberCount,
+				r.Channel.CreatedBy.ID,
+				r.Channel.CreatedAt.Format(time.RFC822),
+				r.Channel.UpdatedAt.Format(time.RFC822),
+				r.Channel.ExtraData)
+
+			if ctx.Bool("json") {
+				PrintRawJson(r)
+			} else {
+				t.Print()
+			}
+
+			return nil
+		},
+	}
+}
+
+func newCreateChannelCmd(config *Config) *cli.Command {
+	return &cli.Command{
+		Name:      "create",
+		Usage:     "Create a new channel or return an existing one.",
```

Comment:
What is the need to return the existing channel? I suggest to return an error (or a message) saying that the channel already exists

Reply:
I don't know if we get a warning from a backend if it already exists.

https://getstream.io/chat/docs/rest/#channels-getorcreatechannel_type_id

Reply:
we don't, but we can check with created_at value maybe?

Reply:
✅